e2e/test/framework/pod.go

```diff
@@ -88,7 +104,11 @@ func (f *Invocation) CreatePod(pod *core.Pod) error {
 }
 
 func (f *Invocation) DeletePod(meta metav1.ObjectMeta) error {
-	return f.kubeClient.CoreV1().Pods(meta.Namespace).Delete(meta.Name, deleteInForeground())
+	err := f.kubeClient.CoreV1().Pods(meta.Namespace).Delete(meta.Name, deleteInForeground())
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
Does this version of `IsNotFound` not accept a nil error? If so, why are you using this function in `WaitForDelete` without the `if err != nil` gate?

Reply:
so the bigger problem with WaitForDelete is it takes an ObjectMeta and assumes it's a pod, its supposed to be deleted and I'll remove some unused code in separate commit.

RE the actual question, it is fine to not have the nil check and ill rewrite DeletePod accordingly.